### PR TITLE
adding linters to ekf_localizer

### PR DESCRIPTION
**Pull request by nik-tier4**
`cpp_lint` giving some errors related to TODO.

### CODE REVIEW

File: localization/pose_twist_fusion_filter/ekf_localizer/package.xml [hunk 1]
```diff
@@ -22,6 +22,7 @@
   <test_depend>ament_cmake_gtest</test_depend>
   <test_depend>ament_lint_auto</test_depend>
   <test_depend>ament_lint_common</test_depend>
+  <test_depend>ament_cmake_cppcheck</test_depend>
```
Comment (esteve) [inline localization/pose_twist_fusion_filter/ekf_localizer/package.xml:25]: No need to add `ament_cmake_cppcheck` as it's already part of `ament_lint_common`

File: localization/pose_twist_fusion_filter/ekf_localizer/src/ekf_localizer.cpp [hunk 2]
```diff
@@ -23,44 +27,51 @@
...
 
-#include <functional>
 using std::placeholders::_1;
 
 EKFLocalizer::EKFLocalizer(const std::string & node_name, const rclcpp::NodeOptions & node_options)
 : rclcpp::Node(node_name, node_options), dim_x_(6 /* x, y, yaw, yaw_bias, vx, wz */)
 {
-  show_debug_info_ = declare_parameter("show_debug_info", bool(false));
-  ekf_rate_ = declare_parameter("predict_frequency", double(50.0));
+  show_debug_info_ = declare_parameter("show_debug_info", static_cast<bool>(false));
```
Comment (esteve) [inline localization/pose_twist_fusion_filter/ekf_localizer/src/ekf_localizer.cpp:35]: This cast is not needed.

File: localization/pose_twist_fusion_filter/ekf_localizer/src/ekf_localizer.cpp [hunk 3]
```diff
@@ -23,44 +27,51 @@
...
+  ekf_rate_ = declare_parameter("predict_frequency", static_cast<double>(50.0));
```
Comment (esteve) [inline localization/pose_twist_fusion_filter/ekf_localizer/src/ekf_localizer.cpp:36]: Cast not needed here either.

File: localization/pose_twist_fusion_filter/ekf_localizer/src/ekf_localizer.cpp [hunk 4]
```diff
@@ -23,44 +27,51 @@
...
   ekf_dt_ = 1.0 / std::max(ekf_rate_, 0.1);
-  tf_rate_ = declare_parameter("tf_rate", double(10.0));
-  enable_yaw_bias_estimation_ = declare_parameter("enable_yaw_bias_estimation", bool(true));
-  extend_state_step_ = declare_parameter("extend_state_step", int(50));
+  tf_rate_ = declare_parameter("tf_rate", static_cast<double>(10.0));
```
Comment (esteve) [inline localization/pose_twist_fusion_filter/ekf_localizer/src/ekf_localizer.cpp:38]: Cast

File: localization/pose_twist_fusion_filter/ekf_localizer/src/ekf_localizer.cpp [hunk 5]
```diff
@@ -23,44 +27,51 @@
...
+  enable_yaw_bias_estimation_ =
+    declare_parameter("enable_yaw_bias_estimation", static_cast<bool>(true));
```
Comment (esteve) [inline localization/pose_twist_fusion_filter/ekf_localizer/src/ekf_localizer.cpp:40]: Cast

**Review by esteve — CHANGES_REQUESTED** [review]

**Comment by esteve** [comment]
@nik-tier4 casts for parameters shouldn't be needed. As for the `TODO` in https://github.com/nik-tier4/Pilot.Auto/blob/add_linters_ekf_localizer/localization/pose_twist_fusion_filter/ekf_localizer/src/ekf_localizer.cpp#L324, `git blame` says it was @mitsudome-r who last modified that line. Could you add `TODO(mitsudome-r)` ? Thanks.

**Comment by nik-tier4** [comment]
@esteve all changes made. Kindly review it.

**Review by esteve — APPROVED** [review]
@nik-tier4 thanks for addressing my feedback.

**Pull request merged**